backends/hip-gen/ceed-hip-gen-operator-build.cpp

```diff
@@ -1130,16 +1140,14 @@ extern "C" int CeedHipGenOperatorBuild(CeedOperator op) {
     CeedChkBackend(ierr);
     if (emode==CEED_EVAL_GRAD)
     {
+      code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Q1d];\n";
       if (useCollograd) {
         //Accumulator for gradient slices
-        code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Q1d];\n";
         code << "    for (CeedInt i = 0; i < ncomp_out_"<<i<<"; ++i) {\n";
         code << "      for (CeedInt j = 0; j < Q1d; ++j) {\n";
         code << "        r_tt"<<i<<"[j + i*Q1d] = 0.0;\n";
         code << "      }\n";
         code << "    }\n";
-      } else {
-        code << "    CeedScalar r_tt"<<i<<"[ncomp_out_"<<i<<"*Dim*Q1d];\n";
       }
```

Comment:
Glad that CUDA and HIP are now the same, but the more I think about this, the less I understand this change. 
Why does it no longer matter whether we are using collograd or not when determining the size of the `r_tt` array?  If we used to need the extra `Dim` factor in non-collograd case before, why not now?

Reply:
I'm not completely sure about this change either... I'll revert I think.

Reply:
Reverted in [fe667b8](https://github.com/CEED/libCEED/pull/1021/commits/fe667b84b42c45b322bb0a9d91af5da647d51611).